utils/litemap/src/store/mod.rs

```diff
@@ -133,6 +133,24 @@ pub trait StoreMut<K, V>: Store<K, V> {
             }
         }
     }
+
+    /// Extends this store with items from an iterator.
+    fn lm_extend<I>(&mut self, other: I)
```

Comment:
issue: having the quadratic algorithm in the default implementation is a footgun. same for `lm_retain`

Reply:
To clarify, are you suggesting the default impl for lm_extend and lm_retain* to be removed?

* not part of the PR

Reply:
yes

Reply:
(if we're happy to make breaking changes)